// @wonderflow/react-components/dist/components/expander/expander.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { Expander } from './expander';
const story = {
    title: 'Actions/Expander',
    component: Expander,
    args: {
        visibleArea: '100px',
    },
};
export default story;
const Template = args => (_jsxs(Expander, { ...args, children: ["Lorem ipsum dolor sit amet consectetur adipisicing elit. Doloribus corporis nobis ipsum natus laudantium dolorem sapiente, sint adipisci at asperiores quia ex voluptatum veniam! Quos rerum natus eos excepturi atque! Lorem ipsum dolor sit amet consectetur adipisicing elit. Doloribus corporis nobis ipsum natus laudantium dolorem sapiente, sint adipisci at asperiores quia ex voluptatum veniam! Quos rerum natus eos excepturi atque! Lorem ipsum dolor sit amet consectetur adipisicing elit.", _jsx("br", {}), "Doloribus corporis nobis ipsum natus laudantium dolorem sapiente, sint adipisci at asperiores quia ex voluptatum veniam! Quos rerum natus eos excepturi atque! Lorem ipsum dolor sit amet consectetur adipisicing elit.", _jsx("img", { alt: "", src: "https://images.unsplash.com/photo-1527831200629-a0438d65142b?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1740&q=80" })] }));
export const Default = Template.bind({});
export const Open = Template.bind({});
Open.args = {
    defaultExpanded: true,
};
