// @wonderflow/react-components/dist/components/list/list.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { List } from './list';
const story = {
    title: 'Typography/List',
    component: List,
    args: {
        hideMarker: false,
    },
    argTypes: {
        dimension: {
            options: ['small', 'regular', 'big'],
            control: { type: 'radio' },
        },
    },
};
export default story;
const Template = ({ marker, markerColor, ...args }) => (_jsxs(List, { ...args, children: [_jsx(List.Li, { marker: marker, markerColor: markerColor, children: "List item text" }), _jsx(List.Li, { marker: marker, markerColor: markerColor, children: "List item text List item textList item textList item textList item text" }), _jsx(List.Li, { marker: marker, markerColor: markerColor, children: "List item text" })] }));
export const Default = Template.bind({});
export const CustomMarker = Template.bind({});
CustomMarker.args = {
    marker: 'circle-check',
};
export const MarkerColor = Template.bind({});
MarkerColor.args = {
    marker: 'circle-check',
    markerColor: 'var(--highlight-green-foreground)',
};
