// @wonderflow/react-components/dist/components/tooltip/tooltip.stories.js
import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "react/jsx-runtime";
import { Button, Text } from '../..';
import { Tooltip } from './tooltip';
const story = {
    title: 'Dialogs/Tooltip',
    component: Tooltip,
    args: {
        trigger: _jsx(Text, { as: "span", children: "HTML" }),
        interactive: true,
    },
};
export default story;
const Template = args => (_jsxs(_Fragment, { children: [_jsxs(Tooltip, { ...args, children: [_jsx("button", { type: "button", children: "test" }), "HTML stands for HyperText Markup LanguageHTML stands for HyperText Markup LanguageHTML stands for HyperText Markup LanguageHTML stands for HyperText Markup LanguageHTML stands for HyperText Markup Language", _jsx("button", { type: "button", children: "test" })] }), _jsxs(Tooltip, { ...args, children: [_jsx("button", { type: "button", children: "test" }), "HTML stands for HyperText Markup LanguageHTML stands for HyperText Markup LanguageHTML stands for HyperText Markup LanguageHTML stands for HyperText Markup LanguageHTML stands for HyperText Markup Language", _jsx("button", { type: "button", children: "test" })] })] }));
export const Default = Template.bind({});
Default.args = {
    interactive: true,
};
export const WithButtons = Template.bind({});
WithButtons.args = {
    trigger: _jsx(Button, { children: "HTML" }),
    interactive: true,
};
