// @wonderflow/react-components/dist/components/split-button/split-button.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { Menu } from '../..';
import { SplitButton } from './split-button';
const story = {
    title: 'Actions/Split button',
    component: SplitButton,
    args: {
        dimension: 'regular',
        kind: 'primary',
        label: 'Click me',
        fullWidth: false,
        disabled: true,
        busy: false,
        placement: 'auto-start',
        offset: 8,
    },
    argTypes: {
        onClick: {
            action: 'clicked',
            table: {
                disable: true,
            },
        },
        dimension: {
            options: ['small', 'regular', 'big'],
            control: { type: 'radio' },
        },
        kind: {
            options: ['primary', 'secondary', 'flat'],
            control: { type: 'radio' },
        },
    },
};
export default story;
const Template = args => (_jsx(SplitButton, { ...args, children: _jsxs(Menu, { children: [_jsx(Menu.Item, { value: "1", padding: false, children: "Option 1" }), _jsx(Menu.Item, { value: "2", padding: false, children: "Option 2" })] }) }));
export const Default = Template.bind({});
