// @wonderflow/react-components/dist/components/modal/modal.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useState } from 'react';
import { Button, IconButton, OverlayContainer, ResponsiveProvider, Stack, Title, useOverlayContext, } from '../..';
import { Modal } from './modal';
const story = {
    title: 'Dialogs/Modal',
    component: Modal,
    argTypes: {},
    args: {},
};
export default story;
const ModalShell = ({ children, ...otherProps }) => {
    const [isVisible, setIsVisible] = useState(false);
    return (_jsxs(ResponsiveProvider, { children: [_jsx(Button, { onClick: () => setIsVisible(true), children: "Show Modal" }), _jsx(OverlayContainer, { onClose: () => setIsVisible(false), children: isVisible && (_jsx(Modal, { ...otherProps, children: children }, "dynamic-modal")) })] }));
};
const DefaultTemplate = args => (_jsx(ModalShell, { ...args, children: _jsxs(Modal.Content, { title: "Modal title", theme: "auto", children: ["Lorem ipsum dolor sit amet consectetur adipisicing elit. Doloribus et magnam distinctio qui quod ducimus libero magni earum perspiciatis.", _jsx("img", { width: "100%", height: "400", alt: "", src: "https://images.unsplash.com/photo-1579332649290-10b7da0cd111?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=cover&w=1600&q=80" }), _jsx("button", { type: "button", children: "click" })] }) }));
export const Default = DefaultTemplate.bind({});
const CustomContentModal = () => {
    const { onClose, titleId } = useOverlayContext();
    return (_jsxs(Stack, { vAlign: "center", children: [_jsxs(Stack, { direction: "row", fill: false, vAlign: "center", hAlign: "space-between", children: [_jsx(Title, { level: "5", id: titleId, children: titleId }), _jsx(IconButton, { onClick: () => onClose?.(), icon: "xmark", kind: "flat", "aria-label": "Close modal" })] }), "Lorem ipsum dolor sit amet consectetur adipisicing elit. Doloribus et magnam distinctio qui quod ducimus libero magni earum perspiciatis.", _jsx("img", { width: "100%", height: "auto", alt: "", src: "https://images.unsplash.com/photo-1579332649290-10b7da0cd111?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=cover&w=1600&q=80" })] }));
};
const CustomTemplate = args => (_jsx(ModalShell, { ...args, children: _jsx(CustomContentModal, {}) }));
export const Custom = CustomTemplate.bind({});
