// @wonderflow/react-components/dist/components/disclosure/disclosure.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { Disclosure } from './disclosure';
const story = {
    title: 'Actions/Disclosure',
    component: Disclosure,
    args: {
        padding: true,
        expandable: true,
        dimension: 'regular',
        summary: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        open: false,
    },
    argTypes: {
        expandable: {
            options: [true, false],
            control: { type: 'inline-radio' },
        },
        padding: {
            options: [true, false],
            control: { type: 'inline-radio' },
        },
        dimension: {
            options: ['small', 'regular', 'big'],
            control: { type: 'inline-radio' },
        },
    },
};
export default story;
const Template = args => (_jsx(Disclosure, { ...args, children: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Doloribus corporis nobis ipsum natus laudantium dolorem sapiente, sint adipisci at asperiores quia ex voluptatum veniam! Quos rerum natus eos excepturi atque!" }));
export const Default = Template.bind({});
const NestedTemplate = args => (_jsxs(Disclosure, { ...args, children: ["Lorem ipsum dolor sit amet consectetur adipisicing elit. Doloribus corporis nobis ipsum natus laudantium dolorem sapiente, sint adipisci at asperiores quia ex voluptatum veniam! Quos rerum natus eos excepturi atque!", _jsx(Disclosure, { summary: "Nested", children: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Doloribus corporis nobis ipsum natus laudantium dolorem sapiente, sint adipisci at asperiores quia ex voluptatum veniam! Quos rerum natus eos excepturi atque!" })] }));
export const Nested = NestedTemplate.bind({});
