// @wonderflow/react-components/dist/components/title/title.stories.js
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { Container } from '../..';
import { Title } from './title';
const story = {
    title: 'Typography/Title',
    component: Title,
    parameters: {
        controls: { hideNoControlsWarning: true },
    },
    args: {
        responsive: true,
    },
    argTypes: {
        level: {
            options: ['display', '1', '2', '3', '4', '5', '6'],
            control: { type: 'select' },
        },
        lineHeight: {
            options: ['none', 'small', 'large'],
            control: { type: 'inline-radio' },
        },
        textAlign: {
            options: ['start', 'center', 'end'],
            control: { type: 'inline-radio' },
        },
    },
};
export default story;
export const Default = ({ lineHeight }) => (_jsxs(Container, { children: [_jsx(Title, { lineHeight: lineHeight, as: "h1", level: "display", children: "Sample title" }), _jsx(Title, { lineHeight: lineHeight, as: "h1", level: "1", children: "Sample title" }), _jsx(Title, { lineHeight: lineHeight, as: "h2", level: "2", children: "Sample title" }), _jsx(Title, { lineHeight: lineHeight, as: "h3", level: "3", children: "Sample title" }), _jsx(Title, { lineHeight: lineHeight, as: "h4", level: "4", children: "Sample title" }), _jsx(Title, { lineHeight: lineHeight, as: "h5", level: "5", children: "Sample title" }), _jsx(Title, { lineHeight: lineHeight, as: "h6", level: "6", children: "Sample title" })] }));
const Template = args => (_jsx(Container, { dimension: "medium", children: _jsx(Title, { ...args, children: "Sample title" }) }));
export const Single = Template.bind({});
Single.args = {
    as: 'span',
    lineHeight: 'small',
    level: '1',
    maxWidth: 'auto',
    textAlign: 'center',
};
export const WithAnchor = Template.bind({});
WithAnchor.args = {
    as: 'span',
    lineHeight: 'small',
    level: '1',
    maxWidth: 'auto',
    textAlign: 'center',
    anchor: true,
};
